src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers.Tests/AZC0013Tests.cs

```diff
@@ -0,0 +1,174 @@
+    }
+}
+");
+            var diagnostics = await _runner.GetDiagnosticsAsync(testSource.Source);
+            var diagnostic = Assert.Single(diagnostics);
+
+            Assert.Equal("AZC0013", diagnostic.Id);
+            AnalyzerAssert.DiagnosticLocation(testSource.DefaultMarkerLocation, diagnostics[0].Location);
+        }
+
+        [Fact]
+        public async Task AZC0013NoWarningOnExistingConfigureAwaitTrue()
+        {
+            var testSource = TestSource.Read(@"
+namespace RandomNamespace
+{
+    public class MyClass
+    {
+        public static async System.Threading.Tasks.Task Foo()
+        {
+#pragma warning disable AZC0013
+            await System.Threading.Tasks.Task.Run(() => {}).ConfigureAwait(true);
+#pragma warning restore AZC0013
+        }
+    }
+}
+");
+            var diagnostics = await _runner.GetDiagnosticsAsync(testSource.Source);
+            Assert.Empty(diagnostics);
+        }
+
+        [Fact]
+        public async Task AZC0013WarningOnAsyncEnumerableExistingConfigureAwaitTrue()
+        {
+            var testSource = TestSource.Read(@"
+namespace RandomNamespace
+{
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    public class MyClass
+    {
+        public static async Task Foo()
+        {
+            await foreach (var x in GetValuesAsync()./*MM*/ConfigureAwait(true)) { }
+        }
+
+        private static async IAsyncEnumerable<int> GetValuesAsync() { yield break; }
+    }
+}
+");
+            var diagnostics = await _runner.GetDiagnosticsAsync(testSource.Source);
+            var diagnostic = Assert.Single(diagnostics);
+
+            Assert.Equal("AZC0013", diagnostic.Id);
+            AnalyzerAssert.DiagnosticLocation(testSource.DefaultMarkerLocation, diagnostics[0].Location);
+        }
+
+        [Fact]
+        public async Task AZC0013NoWarningOnAsyncEnumerableExistingConfigureAwaitTrue()
+        {
+
+            var testSource = TestSource.Read(@"
+namespace RandomNamespace
+{
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    public class MyClass
+    {
+        public static async Task Foo()
+        {
+#pragma warning disable AZC0013
+            await foreach (var x in /*MM*/GetValuesAsync().ConfigureAwait(true)) { }
+#pragma warning restore AZC0013
+        }
+
+        private static async IAsyncEnumerable<int> GetValuesAsync() { yield break; }
+    }
+}
+");
+            var diagnostics = await _runner.GetDiagnosticsAsync(testSource.Source);
+            Assert.Empty(diagnostics);
+        }
+
+        [Fact]
+        public async Task AZC0013WarningOnAsyncUsingExistingConfigureAwaitTrue()
+        {
+            var testSource = TestSource.Read(@"
+namespace RandomNamespace
+{
+    using System;
+    using System.Threading.Tasks;
+
+    public class MyClass
+    {
+        public static async Task Foo()
+        {
+            var ad = new AsyncDisposable();
```

Comment:
would we catch `await using var ad = AsyncDisposable();` ?

Reply:
Nice catch: https://github.com/Azure/azure-sdk-tools/issues/328